iot/api-client/end_to_end_example/cloudiot_pubsub_example_mqtt_device.py

```diff
@@ -121,7 +121,7 @@ def on_subscribe(self, unused_client, unused_userdata, unused_mid,
 
     def on_message(self, unused_client, unused_userdata, message):
         """Callback when the device receives a message on a subscription."""
-        payload = str(message.payload)
+        payload = message.payload
         print('Received message \'{}\' on topic \'{}\' with Qos {}'.format(
             payload, message.topic, str(message.qos)))
```

Comment:
Should you remove the str(message.qos) from this line too?

Reply:
A string formatter will automatically cast it into string. So, it could be an useless casting, but really doesn't matter. (message.qos is an int type, I guess)

The perfect printing might be like

```python3
print('Received message \'{}\' on topic \'{}\' with Qos {}'.format(
             payload.decode(), message.topic, message.qos))
```

But exposing type of *payload* could be useful to specify that it is a **byte** type.

Therefore, it just depends on the author's intention, just for logging or providing extra information though the log.

Reply:
We can leave as-is.